code/modules/mob/living/silicon/silicon.dm

```diff
@@ -153,6 +153,11 @@
 	return 1
 
 /mob/living/silicon/bullet_act(var/obj/item/projectile/Proj)
+	if(istype(Proj, /obj/item/projectile/energy/electrode/tesla))
```

Comment:
Move siliconstun up to projectile level, then check if the projectile has siliconstun then do the stun, as this could be useful for future silicon-bullying projectiles.